linalg/solvers.cpp

```diff
@@ -3399,6 +3399,52 @@ void ProductSolver::MultTranspose(const Vector & x, Vector & y) const
    y += S0Tz;
 }
 
+OrthoSolver::OrthoSolver() : Solver(0, true), parallel(false) { }
+
+#ifdef MFEM_USE_MPI
+OrthoSolver::OrthoSolver(MPI_Comm mycomm_) : Solver(0, true),
+   mycomm(mycomm_), parallel(true) { }
+#endif
+
+void OrthoSolver::SetOperator(const Operator &op)
+{
+   oper = &op;
+}
+
+void OrthoSolver::Mult(const Vector &b, Vector &x) const
+{
+   // Orthogonalize input
+   Orthogonalize(b, b_ortho);
```

Comment:
We should doble-check if we need this input orthogonalization (the output one is necessary).

If the right-hand side and initial guess is compatible (and the operator has constants in its kernel) everything else in a Krylov solver should remain in the subspace of zero mean values, at least in exact arithmetic.

Reply:
@jandrej could you please check if this is needed in the Navier runs?

Reply:
It is not needed because @tzanio's statement is correct. The initial guess has to be compatible, which is ensured by https://github.com/mfem/mfem/blob/master/miniapps/navier/navier_solver.cpp#L540-L543.